lib/matplotlib/collections.py

```diff
@@ -1843,17 +1846,20 @@ def __init__(self, patches, *, match_original=False, **kwargs):
         """
 
         if match_original:
-            def determine_facecolor(patch):
-                if patch.get_fill():
-                    return patch.get_facecolor()
-                return [0, 0, 0, 0]
-
-            kwargs['facecolors'] = [determine_facecolor(p) for p in patches]
-            kwargs['edgecolors'] = [p.get_edgecolor() for p in patches]
+            kwargs['facecolors'] = [p.get_facecolor() for p in patches]
             kwargs['linewidths'] = [p.get_linewidth() for p in patches]
             kwargs['linestyles'] = [p.get_linestyle() for p in patches]
             kwargs['antialiaseds'] = [p.get_antialiased() for p in patches]
 
+            # Edgecolors are handled separately because are defaulted to None
+            # and the Hatch colors depend on them.
+            if any(p._original_edgecolor is not None for p in patches):
+                kwargs["edgecolors"] = [p.get_edgecolor() for p in patches]
+
+            # Using the hatch of only the first patch
+            if patches:
+                kwargs['hatch'] = patches[0].get_hatch()
```

Comment:
This is still problematic because it broadcasts the hatch of the first patch to all of them, if we are matching original then we need to keep track of which (if any) of the patches were originaly hatched.